From c8a4e106301fba41796f2ae4ca9514be20f6bb97 Mon Sep 17 00:00:00 2001
From: Timothy Gu <timothygu99@gmail.com>
Date: Thu, 19 Mar 2020 23:27:15 -0400
Subject: [PATCH 1/2] tests/fprintd: Allow tests to run even when virtual
 device is not available

Regression since 0fb4f3b0217ac0fe2ac7a65a5ff2abed68d74a53, which
consistently removes self.device rather than setting it to None.
---
 tests/fprintd.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/fprintd.py b/tests/fprintd.py
index 1bbc38b..5d9a329 100755
--- a/tests/fprintd.py
+++ b/tests/fprintd.py
@@ -318,7 +318,7 @@ class FPrintdVirtualDeviceBaseTest(FPrintdTest):
         self.polkitd_start()
         self.daemon_start()
 
-        if self.device is None:
+        if not hasattr(self, 'device'):
             self.skipTest("Need virtual_image device to run the test")
 
         self._polkitd_obj.SetAllowed(['net.reactivated.fprint.device.setusername',
-- 
2.24.1


From 418e1391f760a58c0bae3aeec43e3fa81874b6ba Mon Sep 17 00:00:00 2001
From: Timothy Gu <timothygu99@gmail.com>
Date: Thu, 19 Mar 2020 23:29:42 -0400
Subject: [PATCH 2/2] tests/fprintd: Increase allowed timeout

On my laptop it takes 48 seconds to run, even when virtual device is not
available.
---
 tests/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/meson.build b/tests/meson.build
index 5f84d5b..7e0701d 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -18,6 +18,7 @@ foreach t: tests
             'FPRINT_BUILD_DIR=' + meson.build_root() / 'src',
             'TOPSRCDIR=' + meson.source_root(),
         ],
+        timeout: t == 'fprintd' ? 60 : 30,
     )
 endforeach
 
-- 
2.24.1

